[REVIEW] replacing logging.warn with logging.warning

jameslamb opened the pull request:
:wave: hello from Chicago!

I'm starting to get familiar with `cuml`, and really liking it! I wanted to start contributing back, so I ran `pylint` over this codebase's Python code. Many of the warnings from this tool are subjective style things that are best left to maintainers, but there are some genuine code-correctness things, and I think it would be useful to fix them.

This PR proposes replacing uses of `logging.warn()` with `logging.warning()`. `logging.warn()` [has been deprecated since Python 3.3](https://docs.python.org/3/library/logging.html#logging.Logger.warning), and might be removed in a future Python release.

![image](https://user-images.githubusercontent.com/7608904/96008645-63fbbe00-0e05-11eb-971f-0d63e1108639.png)

Thanks for your time and consideration.

GPUtester commented:
Can one of the admins verify this patch?

GPUtester commented:
Can one of the admins verify this patch?

JohnZed commented:
Thanks for submitting, @jameslamb ! This was thought-provoking in that it reminds us we should totally remove this spmg_ipc file (issue #2979 ). In general, we should be using the new cuml logging code instead of Python's logger. (That allows us to unify C++ and python loggin.) If you're looking for a issue to tackle, you could try ripping out this spmg_ipc.py file entirely

jameslamb commented:
I've updated this pull request to point at `branch-0.17`, and merged the most recent changes from that branch into this one.

ajschmidt8 commented:
add to whitelist

ajschmidt8 commented:
Removing `ops-codeowners` from the required reviews since it doesn't seem there are any file changes that we're responsible for. Feel free to add us back if necessary.

jameslamb commented:
I was looking through my pull request list today and realized this one from October is still open. I've merged in the latest changes from `branch-0.18` and changed this PR so that it now points at `branch-0.18` (was originally pointed at `branch-0.17`).

I did not add a changelog entry in `CHANGELOG.md` because I didn't see any other 0.18 changes there. It looks like they're being updated by maintainers in this draft release: https://github.com/rapidsai/cuml/releases/tag/branch-0.18-latest

This was the changelog entry I previously had:

> - PR #2976: replace `logging.warn()` with `logging.warning()`

Let me know if there's anything else I need to change or test.

jameslamb commented:
> This PR has been labeled `inactive-30d` due to no recent activity in the past 30 days. Please close this PR if it is no longer required. Otherwise, please respond with a comment indicating any updates. This PR will be labeled `inactive-90d` if there is no activity in the next 60 days.

I believe this PR is mergeable. It just needs a review. I just updated it with the latest changes from `branch-0.18` and confirmed with `git grep -E "logging\.warn"` that no new uses of `logging.warn()` have been added since this PR was first opened.

jameslamb commented:
I see that `cuml` is now on `branch-0.19`. I just rebased this PR to that branch and changed the target branch for this pull request.

Is there some additional information reviewers would like me to provide or other changes you'd like to see? This PR was originally opened against `branch-0.16` six months ago and I've been keeping it up to date with changes in this repo over that time period. I'd like to do what I can to see this PR reach a resolution, even if that resolution is that the PR is rejected.

Thanks for your time and consideration.

raydouglass commented:
ok to test

dantegd reviewed (approved):
@jameslamb sorry for the delay in reviewing and merging your PR! It looks good to go except for 1 minor issue regarding styling, the 2 lines following the second change are missing some additional spaces to make them correctly indented:

‘’’

./python/cuml/dask/common/utils.py:83:26: E128 continuation line under-indented for visual indent
./python/cuml/dask/common/utils.py:84:26: E128 continuation line under-indented for visual indent
‘’’

Thanks for the fix! Ill pre approve pending that style fix.

jameslamb commented:
No problem, I just pushed https://github.com/rapidsai/cuml/pull/2976/commits/22441f5d37cde017417c9783fc142e9fb7cae1ea which I think should fix those linting errors.

dantegd commented:
Thanks @jameslamb !

dantegd commented:
@gpucibot merge
